src/neuroglancer/datasource/dvid/frontend.ts

```diff
@@ -253,30 +284,59 @@ export class RepositoryInfo {
   errors: string[] = [];
   dataInstances = new Map<string, DataInstanceInfo>();
   uuid: string;
+  vnodes = new Set<string>();
   constructor (obj: any) {
+    if (obj instanceof RepositoryInfo) {
+      this.alias = obj.alias;
+      this.description = obj.description;
+      // just copy references
+      this.errors = obj.errors;
+      this.dataInstances = obj.dataInstances; 
+      return;  
+    } 
     verifyObject(obj);
     this.alias = verifyObjectProperty(obj, 'Alias', verifyString);
     this.description = verifyObjectProperty(obj, 'Description', verifyString);
     let dataInstanceObjs = verifyObjectProperty(obj, 'DataInstances', verifyObject);
     for (let key of Object.keys(dataInstanceObjs)) {
```

Comment:
Since Object.keys(dataInstanceObjs) constructs a new array, it might be better to call it just once and use it both for the loop and for the calls to parseDataInstance.